text_handler.py

```diff
@@ -26,42 +29,58 @@ async def __handleYoutubeDownload(self, message: types.Message):
         if not str(message.chat.id) in config.WHITELIST_CHAT_ID:
             return
         try:
-            yt = YouTube(message.text)
-            author = yt.author
-            length = yt.length
-            title = yt.title
-            stream = (yt.streams
-                      .filter(only_audio=True, file_extension='mp4')
-                      .order_by('bitrate').desc().first())
-            filesize = stream.filesize
-            video_id = extract.video_id(message.text)
-            video_url = "youtu.be/" + video_id
+            ydl_opts = {
+                    'writethumbnail': True,
+                    'outtmpl': 'downloads/%(extractor)s_%(id)s.%(ext)s',
+                    'format': 'bestaudio',
+            }
+            with YoutubeDL(ydl_opts) as ydl:
+                info = ydl.extract_info(message.text, download=False)
+                audio_url = format(info['url'])
             telegram_audio_limit = 52428800
+            request_head = Request(audio_url, method='HEAD')
+            audio_filesize = urlopen(request_head).headers['Content-Length']
```

Comment:
this should work perfectly for soundcloud, but as for youtube will **Content-Length** return the file size of a video?

Reply:
yeah, I'm sure it returns the audio file size instead of video. because `ydl_opts` has the selection `'format': 'bestaudio'` so the audio_url (`info['url']`) should be an url of an audio file.